### PR TITLE
config_tools: update the regex range of online cpu

### PR DESCRIPTION
**Pull request by Kunhui-Li**
update the regex range of online cpu from [1-99] to [0-9]
in script to make power on/off RTVM normally.

Tracked-On: #6482
Signed-off-by: Kunhui-Li <kunhuix.li@intel.com>

### CODE REVIEW

**Review by NanlinXie — APPROVED** [review]

**Comment by jsun26intel** [comment]
need double check tr usage, minggui says [0-9] is enough.

**Comment by gvancuts** [comment]
> update the regex range of online cpu from [1-99] to [0-9]
> in script to make power on/off RTVM normally.

How does that help with the power management of RTVMs?

**Comment by Kunhui-Li** [comment]
If we use the pattern [1-99], the response of the command "vcpu_list" is as follows. cpu1/10 is misconfigured. Then sos will be crash.

ACRN:\>vcpu_list
VM ID    PCPU ID    VCPU ID    VCPU ROLE    VCPU STATE    THREAD STATE
=====    =======    =======    =========    ==========    ==========
  0         0          0       PRIMARY      Running          RUNNABLE
  0         2          2       SECONDARY    Running          BLOCKED
  0         3          3       SECONDARY    Running          BLOCKED
  0         4          4       SECONDARY    Running          BLOCKED
  0         5          5       SECONDARY    Running          BLOCKED
  0         6          6       SECONDARY    Running          BLOCKED
  0         7          7       SECONDARY    Running          BLOCKED
  **0         10         10      SECONDARY    Running          RUNNABLE**
  0         12         12      SECONDARY    Running          BLOCKED
  0         13         13      SECONDARY    Running          BLOCKED
  0         14         14      SECONDARY    Running          BLOCKED
  0         15         15      SECONDARY    Running          BLOCKED
  2         8          0       PRIMARY      Running          RUNNING
  2         9          1       SECONDARY    Running          RUNNING
  **2         10         2       SECONDARY    Running          RUNNING**
  2         11         3       SECONDARY    Running          RUNNING
 
The below is the response of the command "vcpu_list" after updating the pattern to [0-9]. 
Power on/off RTVM is fine.

ACRN:\>vcpu_list
VM ID    PCPU ID    VCPU ID    VCPU ROLE    VCPU STATE    THREAD STATE
=====    =======    =======    =========    ==========    ==========
  0         0          0       PRIMARY      Running          BLOCKED
  0         1          1       SECONDARY    Running          BLOCKED
  0         2          2       SECONDARY    Running          BLOCKED
  0         3          3       SECONDARY    Running          BLOCKED
  0         4          4       SECONDARY    Running          BLOCKED
  0         5          5       SECONDARY    Running          BLOCKED
  0         6          6       SECONDARY    Running          BLOCKED
  0         7          7       SECONDARY    Running          BLOCKED
  0         12         12      SECONDARY    Running          BLOCKED
  0         13         13      SECONDARY    Running          BLOCKED
  0         14         14      SECONDARY    Running          BLOCKED
  0         15         15      SECONDARY    Running          BLOCKED
  2         8          0       PRIMARY      Running          RUNNING
  2         9          1       SECONDARY    Running          RUNNING
  **2         10         2       SECONDARY    Running          RUNNING**
  2         11         3       SECONDARY    Running          RUNNING

**Comment by gvancuts** [comment]
Thanks, it took me a little while to figure this out (also because I really wasn't familiar with `tr` at all). After experimenting a bit I see what it was doing wrong and I understand how the patch solves the problem.

One more question (and a request below): by default `/sys/devices/system/cpu/cpu0/online` does not exist. I _think_ it could exist in case `CONFIG_BOOTPARAM_HOTPLUG_CPU0` is turned on in the kernel. With this change, isn't there now a risk that we try to offline `cpu0` (which I udnerstand could be problematic)?

Also, the commit message really does not describe what the problem was, nor how it solves it. It does describe a side-effect that was observed but it's really hard for the user to link the two. Can you reword the commit message to make clear what the problem was and how it's solved?

**Comment by Kunhui-Li** [comment]
When to use default setting in the kernel and add cpu0 in offline cpu list by manual, it's no issue by the launch script.
When to turn on `CONFIG_BOOTPARAM_HOTPLUG_CPU0` in the kernel and add cpu0 in offline cpu list by manual, we can't boot to RTVM successfully by the launch script (the log is as follows). And the script has been updated to avoid the risk.

ACRN:\>vcpu_list

VM ID    PCPU ID    VCPU ID    VCPU ROLE    VCPU STATE    THREAD STATE
=====    =======    =======    =========    ==========    ==========
  0         0          0       PRIMARY      Running          RUNNING
  0         1          1       SECONDARY    Running          BLOCKED
  0         2          2       SECONDARY    Running          BLOCKED
  0         3          3       SECONDARY    Running          BLOCKED
  0         4          4       SECONDARY    Running          BLOCKED
  0         5          5       SECONDARY    Running          BLOCKED
  0         6          6       SECONDARY    Running          BLOCKED
  0         7          7       SECONDARY    Running          BLOCKED
  0         12         12      SECONDARY    Running          BLOCKED
  0         13         13      SECONDARY    Running          BLOCKED
  0         14         14      SECONDARY    Running          BLOCKED
  0         15         15      SECONDARY    Running          BLOCKED
  2         8          0       PRIMARY      Running          RUNNING
  2         9          1       SECONDARY    Init             BLOCKED
  2         10         2       SECONDARY    Init             BLOCKED
  2         11         3       SECONDARY    Init             BLOCKED
ACRN:\>

**Review by gvancuts — CHANGES_REQUESTED** [review]
Please update the commit message and it will be good to go!

**Comment by Kunhui-Li** [comment]
The commit message has been updated.

**Comment by gvancuts** [comment]
> 
> 
> The commit message has been updated.

It's still very hard to parse correctly because it does not describe clearly what the problem was, nor how it's fixed. It has nothing to do with ADL for example, it just happens that it has more than 10 CPUs and it triggered the bug. Take a look at my proposal above (and add the explanation about the extra check for `cpu0` to it).

**Comment by Kunhui-Li** [comment]
The commit message has been updated  according to the comment, [#6483 (comment)](https://github.com/projectacrn/acrn-hypervisor/pull/6483#issuecomment-906283276) and also add the explanation about the extra check for cpu0 to it. Please review, thanks.

**Review by gvancuts — APPROVED** [review]
Thanks!

**Comment by NanlinXie** [comment]
@wenlingz , could we remove the "on hold" label and merge it?
